### PR TITLE
Reconcile labels and annotations for configuration/route

### PR DESCRIPTION
**Pull request by hougangliu**
for knative serving,  it reconciles `Labels` of configuration and route, not just spec. refer to [here](https://github.com/knative/serving/blob/master/pkg/reconciler/service/service.go#L414)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/64)

### CODE REVIEW

File: pkg/controller/kfservice/kfservice_controller.go [hunk 1]
```diff
@@ -132,30 +132,30 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	serviceReconciler := ksvc.NewServiceReconciler(r.Client)
 	// Reconcile configurations
-	desiredDefault, desiredCanary := resources.CreateKnativeConfiguration(kfsvc)
```
Comment (ellis-bigelow) [inline pkg/controller/kfservice/kfservice_controller.go]: Can you leave all names in this file? Config is inferred in context.

File: pkg/reconciler/ksvc/reconciler.go [hunk 2]
```diff
@@ -101,6 +102,7 @@ func (c *ServiceReconciler) ReconcileRoute(ctx context.Context, desiredRoute *kn
 	log.Info("Reconciling route diff (-desired, +observed): %s", "diff", diff)
 
 	route.Spec = desiredRoute.Spec
+	route.ObjectMeta.Labels = desiredRoute.ObjectMeta.Labels
```
Comment (ellis-bigelow) [inline pkg/reconciler/ksvc/reconciler.go]: Can we pass annotations too?

**Comment by ellis-bigelow** [comment]
Hey Hougang. Thanks for the fix. Can you follow commit best practices and start with a Capital letter on your commit?

Can you also add a test?

**Comment by yuzisun** [comment]
Thanks @hougangliu! Please also update PR title which is a bit unclear since the fix is to reconcile labels.

File: pkg/controller/kfservice/kfservice_controller.go (hunk 1 above)
Comment (hougangliu) [reply pkg/controller/kfservice/kfservice_controller.go]: Done

File: pkg/reconciler/ksvc/reconciler.go (hunk 2 above)
Comment (hougangliu) [reply pkg/reconciler/ksvc/reconciler.go]: Done

**Comment by hougangliu** [comment]
> Thanks @hougangliu! Please also update PR title which is a bit unclear since the fix is to reconcile labels.

Done

**Comment by hougangliu** [comment]
> Hey Hougang. Thanks for the fix. Can you follow commit best practices and start with a Capital letter on your commit?
> 
> Can you also add a test?

Done

File: pkg/controller/kfservice/kfservice_controller.go (hunk 1 above)
Comment (ellis-bigelow) [reply pkg/controller/kfservice/kfservice_controller.go]: Is there a commit missing? I see see "desiredDefaultConfiguration"

**Comment by ellis-bigelow** [comment]
Thanks Hougang! Just one final bit about variable naming.

File: pkg/controller/kfservice/kfservice_controller.go (hunk 1 above)
Comment (hougangliu) [reply pkg/controller/kfservice/kfservice_controller.go]: it's my mistake, I refactor the commit and push it by force

**Comment by hougangliu** [comment]
> Thanks Hougang! Just one final bit about variable naming.

Sorry, I just cannot understand exactly what it mean `I see see "desiredDefaultConfiguration"`

**Comment by ellis-bigelow** [comment]
> > Thanks Hougang! Just one final bit about variable naming.
> 
> Sorry, I just cannot understand exactly what it mean `I see see "desiredDefaultConfiguration"`

You changed all the variable names in kfservice_controller.go. I would prefer to have these unchanged.

**Comment by hougangliu** [comment]
@ellis-bigelow get it! Done

**Comment by ellis-bigelow** [comment]
Great work!
/lgtm
/approve

**Pull request merged**